### PR TITLE
Allow to add new options dynamically to select fields

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
Pull Request for Improvement.
### Summary of Changes

New method `JFormFieldList::addOption()` to allow to add new options to select field dynamicly.
### Testing Instructions
- Code review
- Apply patch on 3.7.x branch
- Add the following code to issues to any backend view with search tools (ex: for articles list, here https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_content/views/articles/view.html.php#L92 )

``` php
$this->filterForm->getField('fullordering', 'list')
    ->addOption('text1', array('value' => 'field ASC'))
    ->addOption('JFEATURED_DESC', array('value' => 'featured DESC'));
```
- Check there are two now options in the searchtools ordering select field
  image
### Documentation Changes Required

Update documentation that teaches how to add form fields dynamically.

### CODE REVIEW

File: libraries/joomla/form/fields/list.php [hunk 1]
```diff
@@ -159,4 +159,29 @@ protected function getOptions()
 
 		return $options;
 	}
+
+	/**
+	 * Method to add a option to the list field.
```
Comment (brianteeman) [inline libraries/joomla/form/fields/list.php]: Pedantic I know but it is
an option

Comment (andrepereiradasilva) [inline libraries/joomla/form/fields/list.php]: one day i will learn this lol

**Comment by RonakParmar** [comment]
I have tested this item :white_check_mark: successfully on 30817cb34dffd47bdb0dec38a7aa8b8b41c5b70c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12392">issues.joomla.org/tracker/joomla-cms/12392</a>.</sub>

**Comment by RonakParmar** [comment]
System Information: 

Joomla! Version = Joomla! 3.7.0-dev Development [ Noether ] 27-June-2016 08:13 GMT 
screen shot 2016-10-12 at 23 26 25
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12392">issues.joomla.org/tracker/joomla-cms/12392</a>.</sub>

**Comment by zero-24** [comment]
I have tested this item :white_check_mark: successfully on 31993e723e6a303662869192860e34069404a0ce<br /><br />Thanks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12392">issues.joomla.org/tracker/joomla-cms/12392</a>.</sub>

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on 31993e723e6a303662869192860e34069404a0ce<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12392">issues.joomla.org/tracker/joomla-cms/12392</a>.</sub>

**Pull request merged**